From 5cb3328bb3304ec5999015fde3487d01ee0997b7 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Thu, 13 Feb 2020 02:02:02 +0200
Subject: [PATCH] [SW-8229] goya2: add todo open tasks left

Add to code open task which are still left to impliment

Change-Id: I6233249fd48104ecb49aaf15816b79907d1f5b7a
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 81b0a83..ba1a65a 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -129,6 +129,8 @@
 	if (fuse_read(EFUSE_SYSCFG_OFFSET,
 		      &arcboot_rom_global.config_fuse) == -1)
 		arcboot_rom_global.config_fuse = EFUSE_SYSCFG_DEFAULT;
+
+	/* TODO: check in pldm if we have access, if yes get waves */
 	if (arcboot_rom_global.config_fuse & EFUSE_SYSCFG_DISABLE_DEBUG_IF_MASK)
 		security_disable_debug_if();
 
diff --git a/common/efuse.c b/common/efuse.c
index 737c33f..9fd46ef 100644
--- a/common/efuse.c
+++ b/common/efuse.c
@@ -29,6 +29,9 @@
 {
 	int i;
 
+	/* TODO: add cold flop bit so we dont init efuse on warm reset */
+	/* TODO: ask suyash to test time of this function */
+	/* TODO: facto 10 of function time ~10mS*/
 	regmap_writel(EFUSE_READ_RANGE_ALL,
 		      mmPSOC_DFT_EFUSE_READ_ADDRESS);
 
diff --git a/common/secureboot.c b/common/secureboot.c
index d0e596b..c1fec8a 100644
--- a/common/secureboot.c
+++ b/common/secureboot.c
@@ -47,6 +47,10 @@
 	unsigned char *test_public_key = test_phase0b_sig -
 					 test_public_key_len;
 	uint32_t fw_version = base_readl((uint32_t)test_phase0b_data);
+
+	/* TODO: remove any global or ROM specific rutines */
+	/* TODO: add depenancie on validate starp if efuse isn't set */
+	/* TODO: gpio46 is validate strap -  0 is to validate*/
 	if ((arcboot_rom_global.config_fuse &
 	     EFUSE_SYSCFG_ENABLE_VALIDATION_MASK) == 0) {
 		return SECUREBOOT_VERIFY_SUCCESS;
diff --git a/common/security.c b/common/security.c
index 8d3ae29..8565b5b 100644
--- a/common/security.c
+++ b/common/security.c
@@ -612,6 +612,7 @@
 	for (hash_key_index = 0; hash_key_index < SECURITY_KEY_HASH_ARRAY_SIZE;
 	     hash_key_index++) {
 		/* store habana key hash in PSOC_SECURITY */
+		/* TODO: validate correct writing to PSOC_SECURITY */
 		regmap_writel(key_hash[hash_key_index],
 			      security_key_soc_location_offset(hash_key_index));
 		/* store key hashes in Scratch PAD Sram */
diff --git a/common/spi_flash.c b/common/spi_flash.c
index df96707..40ea726 100644
--- a/common/spi_flash.c
+++ b/common/spi_flash.c
@@ -101,6 +101,7 @@
 
 void spi_flash_otp_access(uint8_t enable)
 {
+	/* TODO: check on goya/gaudi real flash otp access */
 	if (enable)
 		regmap_writel(FLASH_OTP_CMD_ENTER, mmPSOC_QSPI_DR0);
 	else
